src/sigstore.ts

```diff
@@ -149,3 +147,60 @@ function configureIdentityProviders(
 
   return idps;
 }
+
+// Assembles the AtifactVerificationOptions from the supplied VerifyOptions.
+function collectArtifactVerificationOptions(
+  options: VerifyOptions
+): sigstore.RequiredArtifactVerificationOptions {
+  // The trusted signers are only used if the options contain a certificate
+  // issuer
+  let signers: sigstore.RequiredArtifactVerificationOptions['signers'];
+  if (options.certificateIssuer) {
+    let san: sigstore.SubjectAlternativeName | undefined = undefined;
+    if (options.certificateIdentityEmail) {
+      san = {
+        type: sigstore.SubjectAlternativeNameType.EMAIL,
+        identity: {
+          $case: 'value',
+          value: options.certificateIdentityEmail,
+        },
+      };
+    } else if (options.certificateIdentityURI) {
+      san = {
+        type: sigstore.SubjectAlternativeNameType.URI,
+        identity: {
+          $case: 'value',
+          value: options.certificateIdentityURI,
+        },
+      };
+    }
+
+    signers = {
+      $case: 'certificateIdentities',
+      certificateIdentities: {
+        identities: [
+          {
+            issuer: options.certificateIssuer,
+            san: san,
```

Comment:
Do we ignore the sane if undefined? Also will there be a way to verify oids by passing in some options? Would be handy on the registry side but not blocking for this PR.

Reply:
At the moment, if you specify an issuer you must also specify a SAN (I added a test to reflect this). 

I've been struggling a bit to decide how strict to be here. The client spec mandates that at least one issuer identity MUST be specified/verified (both issuer AND SAN), but I've attempted to loosen this a bit until we fully understand how we're going to identify the correct values to use when invoking from the CLI and the registry. Definitely open to suggestions.

Also, I do have the internal logic to do the OID-based verification of cert extensions, I just haven't exposed that in the public interface yet. If that will be useful to you in the registry I'll circle back and add that in a subsequent PR.

Reply:
> The client spec mandates that at least one issuer identity MUST be specified/verified (both issuer AND SAN),

hm yeah feels like this would need some more thought in CLI! If the CLI knew the supported issuers, maybe through a registry endpoint, we could use this to drive the validation and maybe cobble together the san from bits in the provenance until we have some way for users to specify this in policies or rules but feels very brittle 🤔 

It makes sense to enforce this when verifying a single bundle using something like cosign but a lot harder at npm scale!